### PR TITLE
Implement those classes that are needed for starting a new game and adding players (resolves #23)

### LINKED ISSUE #23: Start Implementing Dixit Game Logic
You should focus on those classes that are needed for starting a new game and adding players

### PR DESCRIPTION
**Pull request by matyimat**
I don't implement fully this classes, only the necessary parts which have needed to resolve this task. (#23)

### CODE REVIEW

File: Dixit_Logic/Classes/GameState.cs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    /// <summary>
+    /// This class describes a state of game. Every property have internal set method
+    /// so only the logic can modify the game state.
+    /// </summary>
+    class GameState : IGameState
+    {
+        /// <summary>
+        /// Store the player who start the actual turn.
+        /// </summary>
+        private IPlayer _actPlayer;
```
Comment (VSZM) [inline Dixit_Logic/Classes/GameState.cs]: suggestion: change the variable name to storyTeller

File: Dixit_Logic/Classes/GameState.cs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+        /// <summary>
+        /// Store the deck which contains the cards what the players put on the table.
+        /// </summary>
+        private IDeck _boardDeck;
+
+        /// <summary>
+        /// Store a text what describes the association of turn's starter card.
+        /// </summary>
+        private string _cardAssocText;
```
Comment (VSZM) [inline Dixit_Logic/Classes/GameState.cs]: This could be called storyText or simply story

File: Dixit_Logic/Classes/GameState.cs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+        /// <summary>
+        /// Store all players who playe in the actual game. 
+        /// </summary>
+        private IList<IPlayer> _players;
+
+        /// <summary>
+        /// Store a dictionary what associate a palyer with his/her own game-point value.
+        /// </summary>
+        private Dictionary<IPlayer, int> _points;
```
Comment (VSZM) [inline Dixit_Logic/Classes/GameState.cs:54]: I think this could even go to the IPlayer.

File: Dixit_Logic/Classes/Player.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        // override object.Equals
+        public override bool Equals(object obj)
+        {
+            if (obj == null || GetType() != obj.GetType())
+            {
+                return false;
+            }
+
+            Player p = (Player)obj;
+            return (_id == p._id) && (_name == p._name);
```
Comment (VSZM) [inline Dixit_Logic/Classes/Player.cs]: Since ids are unique, Comparing the ID would be enough.

File: Dixit_Logic/Classes/Player.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        }
+
+        /// <summary>
+        /// Make a hash code basd on the identifier and name has code.
+        /// </summary>
+        /// <returns>A hash code</returns>
+        // override object.GetHashCode
+        public override int GetHashCode()
+        {
+            return _id.GetHashCode() ^ _name.GetHashCode();
```
Comment (VSZM) [inline Dixit_Logic/Classes/Player.cs]: Again, just base hashcode on ID. KISS

**Review by VSZM — CHANGES_REQUESTED** [review]

File: Dixit_Logic/Classes/GameState.cs (hunk 3 above)
Comment (matyimat) [reply Dixit_Logic/Classes/GameState.cs:54]: Yes it could , i did it too. But Marci told me that it will be better in practice.

Comment (VSZM) [reply Dixit_Logic/Classes/GameState.cs:54]: Okay, fine by me. Focus on the other points pls.

**Review by VSZM — APPROVED** [review]

**Pull request merged**